### ISSUE TITLE
Steam broadcast not working

### ISSUE THREAD

**Issue by wipe2238** [Pathoschild/SMAPI#338]
When launching SMAPI via Steam - after changing SV properties - it's not possible to broadcast anymore; viewer's client reports 'broadcast not available' or all you see is a black screen. I'm guessing Steam tries to show SMAPI console window instead of game, and fails even in that :)
Everything works perfectly fine when running with default settings.

Low priority, doesn't affect game itself, etc. but it would be nice to see a fix (if it's even possible) in some undefined future.

**Comment by Pathoschild** [comment]
Unfortunately I don't have any experience with Steam broadcasting, but it may be related to #222 which only affects some players. Feel free to post any new information or solutions you find.

**Comment by wipe2238** [comment]
TLDR: changing SMAPI to winforms app and commenting [this little bastard](https://github.com/Pathoschild/SMAPI/blob/7976df856578045b87e17a3d0ea5074715445622/src/StardewModdingAPI/Program.cs#L194) is all what's needed for a quick/dirty fix. 

It worked until i tried to open console "manually" (_AllocConsole()_) via mod, in attempt to restore current functionality; Steam started to throw errors on both sides then, so that's not a way... Then i tried same thing with winform and it changed nothing; sounds heard, nothing visible. Sigh. Looks Steam really hates when more than one window is open at once...

Playing with game window as the only one opened seems to be only thing which actually works. I doubt it's planned to move everything ingame, so... If SMAPI would open console only when some option **is not** present, everything would work as it works currently. If option **is** present, console is never opened (_no-terminal_ looks like a good candidate for that) and some related stuff would need extra checks (looking at you, _Console::set_Title()_).

Personally, i'd be happy already if things above are added, but for these who still want to access log/commands during broadcast, there could be few helpers exposed to mods here and there. Event on new line in log, maybe access to lines before mod's _Entry()_, etc. As for commands, i have no clue; didn't saw a mod with text input yet.

**Comment by Pathoschild** [comment]
Making the console window optional unfortunately isn't feasible, since the console window is SMAPI itself. I haven't seen any other reports for this issue, and I have no current plans to work on this, so I'll mark this as _won't fix_. Note that pull requests are still welcome to fix Steam Broadcast if needed.
